lib/mayaHydra/mayaPlugin/renderOverride.cpp

```diff
@@ -134,6 +135,19 @@ void replaceSelectionTask(PXR_NS::HdTaskSharedPtrVector* tasks)
     *found = HdTaskSharedPtr(new Fvp::SelectionTask);
 }
 
+// We compare the current and previous viewport display style and 
+// return true if we need to recreate the filtering scene indices chain because of a change, false otherwise.
+bool NeedToRecreateTheSceneIndicesChain(unsigned int currentDisplayStyle, unsigned int previousDisplayStyle)
+{
+    const bool currentlyUsingBBoxDisplayStyle = currentDisplayStyle & MHWRender::MFrameContext::kBoundingBox;
+    const bool previouslyUsingBBoxDisplayStyle = previousDisplayStyle & MHWRender::MFrameContext::kBoundingBox;
+    if (currentlyUsingBBoxDisplayStyle != previouslyUsingBBoxDisplayStyle){
+        return true;
+    }
+
+    return false;
```

Comment:
Nit-pick: can be simply `return currentlyUsingBBoxDisplayStyle != previouslyUsingBBoxDisplayStyle;`

Reply:
Makes sense.